request_test.go

```diff
@@ -71,6 +72,19 @@ func TestNewRequest(t *testing.T) {
 	if !reflect.DeepEqual(req.MessageImprint.HashAlgorithm.Parameters, asn1NullRawValue) {
 		t.Fatalf("expected %v, but got %v", asn1NullRawValue, req.MessageImprint.HashAlgorithm.Parameters)
 	}
+
+	defaultRandReader := rand.Reader
+	rand.Reader = &dummyRandReader{}
```

Comment:
nit:
```suggestion
	rand.Reader = &dummyRandReader{}
	defer func() {
		rand.Reader = defaultRandReader
	}
```
Consider this safer pattern.

Reply:
Good idea. Updated.